### ISSUE TITLE
Errors and recommendations

### ISSUE THREAD

**Issue by ArmorKeik** [Kataiser/tf2-rich-presence#195]
Hello, from a translator I am going to give you the errors that I saw in this mod. But I will also give recommendations for the mod.

**1.** The menu image does not match the rest of the map or game mode images. I recommend doing the same as they did in the Mirrorcult mod (https://github.com/mirrorcult/tf2-discord).

![TF2 Rich Presence (Menú)](https://user-images.githubusercontent.com/87909542/132896112-31e927e8-8bc8-4e02-aae1-3fb2e2518214.png)

**2.** Rich Presence does not detect that you are in the menu when you are no longer playing on a server.
It only changes when you connect to another server.

![TF2 Rich Presence (Server)](https://user-images.githubusercontent.com/87909542/132896809-fcfd9d36-ebd4-4495-b7c3-2d5f40f0347e.png)
![TF2 Rich Presence (Other server)](https://user-images.githubusercontent.com/87909542/132896823-cee7973f-156c-4546-8f72-afe6fc2a8a00.png)

**3.** I recommend not translating the community game modes, as very literal and forced translations can be generated (as is the case with the "Deathrun" game mode).

![TF2 Rich Presence (Forced translation)](https://user-images.githubusercontent.com/87909542/132896182-fe52d630-064a-4e0b-92bd-81cd5d74a796.png)

**4.** I recommend, if possible, adding interactive frames in the Rich Presence, in the style of the ROBLOX ItzBlinkzy mod (https://github.com/ItzBlinkzy/roblox-rpc).
They can be boxes that take you to your Steam Profile and another to the server where you are connected; but there is also an option to block this last feature for those who do not want to share the server).

![Roblox RPC mod](https://user-images.githubusercontent.com/87909542/132896259-f722256d-c078-4830-b2e8-b6974249c497.png)

**5.** And finally, add the server name you are playing on, in the style of the previous TF2 RIch Presence mod from Mirrorcult that I passed.

![image](https://user-images.githubusercontent.com/87909542/132896644-d8a724d1-d9fc-4a7e-879d-a189d572373a.png)

**Comment by Kataiser** [comment]
Yo thanks for the detailed feedback! Love to see it.

1. I know it's sized weirdly but I remember not liking the look of it when cropped. I'm also not a fan of the plain BG.
2. Yeah that can happen sometimes when you're playing on community servers, and is guaranteed to happen if you start to load into one but then cancel, neither of which I know how to fix. Let me know if the former is happening *consistently* for you, cause I'd like to fix that.
3. Okay makes sense
4. Ooh interesting, I'll see if I can manage that. No promises, I don't even know if the RPC library I use supports that.
5. I'll look into it

**Comment by ArmorKeik** [comment]
Hello Kataiser. Yes the problem occurs frequently, perhaps always.

Today I entered three different servers, and when I left all of them at no time did it detect that it was on the menu. It kept saying it was on the server.

The information was only updated when I entered another community server, but if I left the menu, it kept marking that it was on the server.

[TF2RP_IG-88_User_v2.0_10.log](https://github.com/Kataiser/tf2-rich-presence/files/7163624/TF2RP_IG-88_User_v2.0_10.log)

**Comment by Kataiser** [comment]
Okay finally got around to looking at this. Detecting leaving still works on my end, and your log looks fairly normal other than changing server IP abruptly like you said. Would you mind sending me the file at `"Steam\steamapps\common\Team Fortress 2\tf\console.log"`, as it is while you're in menus after having left a community server?

**Comment by ArmorKeik** [comment]
Well, here is the text you asked for.

In case it serves you, I will let you know the procedure I did.

In case it serves you, I will let you know the procedure I did.

I ran TF2 **>** I joined a community server **>** I stayed a few seconds inside **>** I pressed Escape and then I gave it to disconnect from the server **>** I selected another community server **>** I waited a few seconds and I disconnected as I did before **>** I exited the game with the exit button.

[console.log](https://github.com/Kataiser/tf2-rich-presence/files/7190772/console.log)

**Comment by Kataiser** [comment]
Right, I see the issue. Not sure I can fix it but I'll see what can be done.

**Comment by Kataiser** [comment]
Sorry for the delay on this, yeah I really don't think I can fix it. Your game just doesn't seem to give me what I need for some reason.

**Comment by Kataiser** [comment]
Oh lol I just tested it and now it's not working for me either

**Comment by ArmorKeik** [comment]
Hello Kataiser.

Oh, I see. So does that mean the problem doesn't just happen to me?

Is there a solution for this? Because if I remember correctly, this does not happen if you play a sv that you are hosting (from the Create Server option from the menu) or practicing with bots in the individual training option.

**Comment by Kataiser** [comment]
Okay I may have fixed it, try this build https://github.com/Kataiser/tf2-rich-presence/suites/3926793725/artifacts/98268065

**Comment by ArmorKeik** [comment]
Buenas tardes.

That build presents an advance, since in the first and third time I exited an sv it showed that it was in the start menu.

But on the second and fourth he kept telling me that I was still inside the sv.

It is a good advance, I continued like this (I hope the translator has put everything right).

**Comment by Kataiser** [comment]
The translation's a bit rough, pero no se mucho Espanol, asi que. (I recommend [DeepL](https://www.deepl.com/), it's often pretty natural sounding.) Well, the contents of your console.log at any given moment are all I have to use for detecting disconnects, so if you'd like to send me that again in the case of a failed detection it would be appreciated. It doesn't help that Valve has been 
opaquely changing how that stuff works recently, so I have to react to that. Oh and to be clear, send it to me *as it is when you're on the main menu*, not after you've joined another server or closed the game.

**Comment by ArmorKeik** [comment]
Hi, sorry for replying late.

Do you want me to try again but now with the build "Updated menus message because Valve did"?

And if it doesn't work then I'll send you the console log but being in the main menu, and not after closing the game.

**Comment by Kataiser** [comment]
I'm in the middle of making more changes this logic so maybe hold off for a bit

**Comment by Kataiser** [comment]
Okay latest build should be good https://github.com/Kataiser/tf2-rich-presence/suites/3940882894/artifacts/98780986

**Comment by ArmorKeik** [comment]
I was testing the last version that you gave me, and it seems that the error of not detecting the menu was solved.

**Comment by Kataiser** [comment]
Awesome! I definitely wanted it to be fixed before Scream Fortress drops
